src/main/java/uk/ac/ebi/ampt2d/metadata/assemblers/GenericResourceAssemblerImpl.java

```diff
@@ -31,7 +31,8 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class ResourceAssemblerImpl<ENTITY extends BaseEntity> implements ResourceAssembler<ENTITY> {
+public class GenericResourceAssemblerImpl<ENTITY extends BaseEntity, RESOURCE extends Resource<ENTITY>>
+        implements GenericResourceAssembler<ENTITY, RESOURCE> {
```

Comment:
The way this is now written, couldn't this class implement directly `ResourceAssembler`, without the intermediate interface?

Reply:
In the GenericResourceAssembler interface I added a new method toResources which is not available in ResourceAssembler. Hence there is an intermediate interface.

Reply:
I have changed this to extend ResourceAssemblerSupport instead and added StudyResource and AssemblyResource to resolve the generic type issue.